roaringbitmap/src/main/java/org/roaringbitmap/buffer/MappeableBitmapContainer.java

```diff
@@ -1269,27 +1269,26 @@ public int nextSetBit(final int i) {
   }
 
   /**
-   * Find the index of the next unset bit greater or equal to i, returns -1 if none found.
+   * Find the index of the next clear bit greater or equal to i.
    *
    * @param i starting index
-   * @return index of the next unset bit
+   * @return index of the next clear bit
    */
-  public short nextUnsetBit(final int i) {
-    int x = i / 64;
+  public int nextClearBit(final int i) {
+    int x = i >> 6; // i / 64 with sign extension
     long w = ~bitmap.get(x);
     w >>>= i;
     if (w != 0) {
-      return (short) (i + numberOfTrailingZeros(w));
+      return i + numberOfTrailingZeros(w);
     }
-    ++x;
-    // for speed, replaced bitmap.limit() with hardcoded MAX_CAPACITY / 64
-    for (; x < MAX_CAPACITY / 64; ++x) {
-      long X = bitmap.get(x);
-      if (X != ~0L) {
-        return (short) (x * 64 + numberOfTrailingZeros(~X));
+    int length = bitmap.limit();
+    for (++x; x < length; ++x) {
+      long map = ~bitmap.get(x);
+      if (map != 0L) {
+        return x * 64 + numberOfTrailingZeros(map);
       }
     }
-    return -1;
+    return length * 64;
```

Comment:
The maximum possible value in the container is the constant `0xFFFF`, and I wonder if C2 would figure out that bitmap.limit() really does always return 1024 in this case.